### PR TITLE
[create-vm-message] Added changes condition

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

Adds a condition to display the `create-vm` helper message on PRs.
Now it is displayed only when a go file is changed.

### CODE REVIEW

File: .gitlab/notify/notify.yml [hunk 1]
```diff
@@ -64,8 +64,12 @@ notify_github:
     - !reference [.except_mergequeue]
     - !reference [.except_main_or_release_branch]
     - !reference [.except_no_tests_no_deploy]
-    - !reference [.on_a7]
```
Comment (amenasria) [inline .gitlab/notify/notify.yml:67]: **❓ question:** ‏Why is this removed?

File: .gitlab/notify/notify.yml [hunk 2]
```diff
@@ -64,8 +64,12 @@ notify_github:
...
-    - when: on_success
+    - changes:
+        paths:
+          - '**/*.go'
+        compare_to: main # TODO: use a variable, when this is supported https://gitlab.com/gitlab-org/gitlab/-/issues/369916
```
Comment (amenasria) [inline .gitlab/notify/notify.yml:70]: **💭 thought:** ‏Crazy that it doesn't exist yet aha

**Review by amenasria — APPROVED** [review]
**👏 praise:** ‏Thanks for this !

Few comments, otherwise LGTM !

File: .gitlab/notify/notify.yml (hunk 1 above)
Comment (CelianR) [reply .gitlab/notify/notify.yml:67]: Otherwise the `.on_a7` triggers the job, I'll try to rewrite it

File: .gitlab/notify/notify.yml [hunk 3]
```diff
@@ -64,8 +64,13 @@ notify_github:
     - !reference [.except_mergequeue]
     - !reference [.except_main_or_release_branch]
     - !reference [.except_no_tests_no_deploy]
-    - !reference [.on_a7]
-    - when: on_success
+    - if: $RELEASE_VERSION_7 != ""
```
Comment (CelianR) [inline .gitlab/notify/notify.yml:67]: @amenasria is it better like this ? I'm not sure I can merge more properly .on_a7 and changes
